src/ui/shadow-scroller/shadow-scroller.hook.ts

```diff
@@ -0,0 +1,136 @@
+import { RefObject, useEffect } from 'react'
+
+import { ScrollDirection } from '@core/hooks/use-grab-scroll'
+import {
+  createShadow,
+  createTrigger,
+  DisplayValue,
+  hide,
+  Placement,
+} from '@ui/shadow-scroller/shadow-scroller.utils'
+
+export const useScrollShadows = (
+  ref: RefObject<HTMLDivElement>,
+  shadowsRef: RefObject<HTMLDivElement>,
+  direction: ScrollDirection,
+  hideShadows: boolean,
+) => {
+  useEffect(() => {
+    const scrollable = ref.current
+    const shadows = shadowsRef.current
+
+    if (!scrollable || !shadows) {
+      return
+    }
+
+    const area = scrollable?.firstElementChild as HTMLDivElement
+
+    if (!area) {
+      return
+    }
+
+    const topTrigger = createTrigger(area, Placement.top)
```

Comment:
i think, we can avoid to create some shadows and triggers based on `direction` variable